ir/value.cpp

```diff
@@ -85,10 +87,12 @@ StateValue Input::toSMT(State &s) const {
   expr undef = expr::mkVar(uname.c_str(), bw);
   s.addUndefVar(undef);
 
-  return { expr::mkIf(type == expr::mkUInt(0, 2),
-                      expr::mkVar(getName().c_str(), bw),
-                      move(undef)),
-           type.extract(1,1) == expr::mkUInt(0, 1) };
+  return {expr::mkIf(config::disable_undef_input ? true
+                                                 : type == expr::mkUInt(0, 2),
+                     expr::mkVar(getName().c_str(), bw), move(undef)),
```

Comment:
Things are a bit more complicated than this.
This encoding will confuse the counterexample printer since it assumes that 1x is poison. If poison is disabled, you may have a counterexample with e.g. type=11, which here will mean undef not poison.
Not sure what's the best way to fix, though. We could inject a precondition to ignore types that are disabled, for example. Or make the counterexample printing less hard-coded. (then this variable could be switched to 1-bit when one of the options is disabled)

Reply:
i injected couple of preconditions that prevents the type variable now to take the disabled value. See the updated patch, please.

Reply:
Latest patch should fix this issue. can you please have a look?